### PR TITLE
Sravan branch

### PR DESCRIPTION
**Pull request by SravanKumarL**

### CODE REVIEW

File: src/client/components/UI/MultiSelect/MultiSelect.js [hunk 1]
```diff
@@ -158,58 +159,13 @@ const styles = theme => ({
...
-//         placeholder: label,
-//         instanceId: 'react-select-single',
-//         id: 'react-select-single',
-//         name: 'react-select-single',
-//         simpleValue: true,
-//         options: props.options,
-//       }}
-//       // InputLabelProps={{
-//       //   shrink: true,
-//       //   className: classes.textFieldFormLabel,
```
Comment (alurakhil24) [inline src/client/components/UI/MultiSelect/MultiSelect.js]: Please remove commented code

File: src/client/components/UI/DataGrid/dataGrid.js [hunk 2]
```diff
@@ -31,6 +33,7 @@ import constants from '../../../../store/sagas/constants';
 import LoadingGrid from './loadingGrid';
 import { Typography, Button, TextField } from 'material-ui';
 import PasswordCell from '../TextField/PasswordCell';
+import phe from 'print-html-element';
```
Comment (alurakhil24) [inline src/client/components/UI/DataGrid/dataGrid.js]: For better readability use a meaningful name

File: src/client/components/UI/DataGrid/dataGrid.js [hunk 3]
```diff
@@ -218,10 +208,9 @@ class DataGrid extends React.PureComponent {
     this.changePageSize = pageSize => this.setState({ pageSize });
     this.commitChanges = ({ added, changed, deleted }) => {
       // if(Object.keys(changed).some(prop=>changed[prop]===''))
-      // {
-      //   this.setState({initialRender:true});
       //   return; // To display snackbar
-      // }
+      if(Object.keys(changed).some(prop=>changed[prop]===undefined))
```
Comment (alurakhil24) [inline src/client/components/UI/DataGrid/dataGrid.js]: if you are trying to check if the prop exists as a key for changed, you could do it this way.. 
if (someProp in changed) then do so and so.. It works the same way as the above.

File: src/client/components/UI/DataGrid/dataGrid.js [hunk 4]
```diff
@@ -119,17 +120,6 @@ const LookupEditCellBase = ({
...
-const PasswordCellBase = ({ value, onValueChange, classes }) => (
-  <TableCell>
-    <TextField type="password" autoComplete="current-password" margin="normal" value={value} onChange={onPasswordChange(onValueChange)} />
-  </TableCell>
-  //className={classes.password}
-);
-const PasswordCell = withStyles(styles, { name: 'DataGrid' })(PasswordCellBase);
 const Cell = (props) => {
   if (props.column.name == "password")
     return (<TableCell>
```
Comment (alurakhil24) [inline src/client/components/UI/DataGrid/dataGrid.js]: Please add all the handler functions before return in functional components and before render in class components.

File: src/client/components/UI/TextField/PasswordCell.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            this.props.onValueChange(e.target.value);
+        };
+    }
+    render() {
+        const { classes } = this.props;
+        return (
+            <TableCell>
+                <TextField type="password" autoComplete="current-password" margin="normal" value={this.state.value} onChange={this.onPasswordChange} />
+            </TableCell>
+            //className={classes.password}
```
Comment (alurakhil24) [inline src/client/components/UI/TextField/PasswordCell.js]: Please make sure you remove all your commented code before you checkin. i would start refactoring with the assumption that you already removed all the commented code in your respective components.

File: src/axios/poojas.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import axios from 'axios';
 
 const instance = axios.create({
-  // baseURL: process.env.SERVER_URI ? process.env.SERVER_URI : 'http://localhost:7000',
-  baseURL: 'https://temple-api-mongo.herokuapp.com',
+  baseURL: process.env.SERVER_URI ? process.env.SERVER_URI : 'http://localhost:7000',
```
Comment (alurakhil24) [inline src/axios/poojas.js]: I tried this but it wasnt taking the environment variable here for some reason. Will look into this. Check in this anyway

File: src/client/components/UI/DataGrid/dataGrid.js [hunk 7]
```diff
@@ -124,10 +127,12 @@ const LookupEditCellBase = ({
   );
 export const LookupEditCell = withStyles(styles, { name: 'DataGrid' })(LookupEditCellBase);
 const Cell = (props) => {
```
Comment (alurakhil24) [inline src/client/components/UI/DataGrid/dataGrid.js]: Destructuring used props woudl be a better option for readability. I would be doing the same for other components

File: src/store/sagas/entity.js [hunk 8]
```diff
@@ -107,7 +107,7 @@ export function* handleFetchData(action) {
 export function* handleFetchSchema(action) {
     const { collection, searchCriteria } = action.payload;
     try {
-        yield put(actions.onFetchReq());
+        yield put(actions.onFetchReq(collection));
         const token = localStorage.getItem('token');
```
Comment (alurakhil24) [inline src/store/sagas/entity.js]: you can select it from state instead using yeild select. please look into it.. These changes will be incorporated soon

File: src/client/components/UI/DataGrid/dataGrid.js [hunk 9]
```diff
@@ -29,7 +31,9 @@ import CancelIcon from 'material-ui-icons/Cancel';
 import { withStyles } from 'material-ui/styles';
 import constants from '../../../../store/sagas/constants';
 import LoadingGrid from './loadingGrid';
-import { Typography, Button } from 'material-ui';
+import { Typography, Button, TextField } from 'material-ui';
+import PasswordCell from '../TextField/PasswordCell';
+import phe from 'print-html-element';
 
 const styles = theme => ({
   lookupEditCell: {
```
Comment (alurakhil24) [inline src/client/components/UI/DataGrid/dataGrid.js]: Please remove this. Its empty

File: src/client/components/UI/DataGrid/dataGrid.js [hunk 10]
```diff
@@ -88,11 +88,11 @@ const commandComponents = {
   cancel: CancelButton,
 };
 
-const Command = ({ id, onExecute,collection }) => {
+const Command = ({ id, onExecute, collection,self }) => {
```
[the remark this answers is not in the document]
Comment (SravanKumarL) [reply src/client/components/UI/DataGrid/dataGrid.js]: sure thing will do

File: src/client/components/UI/DataGrid/dataGrid.js (hunk 2 above)
Comment (SravanKumarL) [reply src/client/components/UI/DataGrid/dataGrid.js]: ok done

File: src/client/components/UI/DataGrid/dataGrid.js (hunk 3 above)
Comment (SravanKumarL) [reply src/client/components/UI/DataGrid/dataGrid.js]: ok I vl do it that way.

File: src/client/components/UI/DataGrid/dataGrid.js (hunk 4 above)
Comment (SravanKumarL) [reply src/client/components/UI/DataGrid/dataGrid.js]: ok done

File: src/client/components/UI/TextField/PasswordCell.js (hunk 5 above)
Comment (SravanKumarL) [reply src/client/components/UI/TextField/PasswordCell.js]: haha ok dude

File: src/axios/poojas.js (hunk 6 above)
Comment (SravanKumarL) [reply src/axios/poojas.js]: nope actually to test it locally I made those changes and reverted em back.

File: src/client/components/UI/DataGrid/dataGrid.js (hunk 7 above)
Comment (SravanKumarL) [reply src/client/components/UI/DataGrid/dataGrid.js]: actually I want to spread em in the child components, So I will look at em anyway

File: src/store/sagas/entity.js (hunk 8 above)
Comment (SravanKumarL) [reply src/store/sagas/entity.js]: interesting one, I will incorporate them :+1:

File: src/client/components/UI/DataGrid/dataGrid.js (hunk 9 above)
Comment (SravanKumarL) [reply src/client/components/UI/DataGrid/dataGrid.js]: sure thing.

File: src/client/components/UI/DataGrid/dataGrid.js [hunk 11]
```diff
@@ -46,11 +46,11 @@ const styles = theme => ({
...
 const AddButton = ({ onExecute, collection }) => (
-  <div style={{width:120}}>
+  <div style={{ width: 120 }}>
     <IconButton onClick={onExecute} title="Create new row">
       <AddIcon />
-      <Typography variant="body2" style={{marginLeft:10}}>
-        Add a new {collection.slice(0,collection.length-1)}
+      <Typography variant="body2" style={{ marginLeft: 10 }}>
+        Add a new {collection.slice(0, collection.length - 1)}
       </Typography>
```
[the remark this answers is not in the document]
Comment (SravanKumarL) [reply src/client/components/UI/DataGrid/dataGrid.js]: lol :joy:

**Comment by SravanKumarL** [comment]
I will raise a new pull request incorporating all these changes

**Pull request closed without merging**